### ISSUE TITLE
Terraform apply fails with a non-default :vpc-id

### ISSUE THREAD

**Issue by stevebuik** [juxt/roll#15]
It complains when creating the ASG that the security group is not in the VPC.

Workaround is to add:

to Machfile.edn after (roll.core/deployment->tf) 

(assoc-in [:resource :aws-security-group "<service name>_443" :vpc_id] (:vpc-id config))

I've heard that using the default VPC in AWS is not a best practice so I switched to my own VPC instead, and hit this problem.

**Comment by jonpither** [comment]
Thanks @stevebuik for reporting. We intend to make Roll fully work with separate VPCs (and to create them automatically) in the short term future, as we are using Roll for a new project that demands this. I will also look into this precise issue asap.

**Comment by stevebuik** [comment]
Good to know. My workaround doesn't completely fix the problem as the ASG cannot start instances correctly. The history shows these errors when each instance starts (and then stops)

Launching a new EC2 instance: i-0aad95c3036b9cc79. Status Reason: Failed to update target group arn:aws:elasticloadbalancing:ap-southeast-2:825788206307:targetgroup/dev-nextdoc-0-alb-tg/487867058ea350e3: The following targets are not in the target group VPC 'vpc-e77d3083': 'i-0aad95c3036b9cc79' Updating load balancer configuration failed

I'm looking into why but I suspect you'll need the answer for your update so I'll update this issue if I find the solution before you do :)

**Comment by stevebuik** [comment]
After spending a half-day without success I'm going to go back to the default VPC and wait for your changes to avoid duplicated work. I was not able to get the full recipe working. I'll follow up on slack but here's what I observed:

- the fix above needed to be applied to both security groups generated by Roll
- instances in non-default VPC don't get ip addresses so aws_launch_configuration needs "associate_public_ip_address" : true
- the target group wasn't able to detect a healthy instance. I suspect this was because of port 8080 access in the non-default subnet/ACL. I couldn't work out how to fix this.

I'd suggest that this will be tricky for everyone so a well documented recipe for creating the custom VPC/subnets/ACL etc will really help. Even better would be a terraform file or even have Roll generate the extra TF json required. Just a thought but I could imagine other AWS beginners having similar problems as I did.
